src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ForbidPrivateEventHandlers/ForbidPrivateEventHandlersFix.cs

```diff
@@ -0,0 +1,133 @@
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Collections.Generic;
+using System.Threading;
+using System.Linq;
+using Microsoft.CodeAnalysis.Text;
+using Acuminator.Utilities.Common;
+
+namespace Acuminator.Analyzers.StaticAnalysis.ForbidPrivateEventHandlers
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+	public class ForbidPrivateEventHandlersFix : PXCodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldNotBePrivate.Id,
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual.Id,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations.Id);
+
+		protected override Task RegisterCodeFixesForDiagnosticAsync(CodeFixContext context, Diagnostic diagnostic)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var registerCodeFix = DiagnosticUtils.IsFlagSet(diagnostic, StaticAnalysis.DiagnosticProperty.RegisterCodeFix);
+			var isContainingTypeSealed = DiagnosticUtils.IsFlagSet(diagnostic, DiagnosticProperty.IsContainingTypeSealed);
+			var addVirtualModifier = DiagnosticUtils.IsFlagSet(diagnostic, DiagnosticProperty.AddVirtualModifier);
+
+			if (!registerCodeFix)
+			{
+				return Task.CompletedTask;
+			}
+
+			var accessibilityModifier = isContainingTypeSealed
+				? SyntaxKind.PublicKeyword
+				: SyntaxKind.ProtectedKeyword;
+
+			var modifierFormatArg = ForbidPrivateEventHandlersAnalyzer.GetModifierFormatArg(accessibilityModifier, addVirtualModifier);
+
+			var makeProtectedTitle = nameof(Resources.PX1077Fix).GetLocalized(modifierFormatArg).ToString();
+
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+
+			var codeFixAction = CodeAction.Create(
+				makeProtectedTitle,
+				cToken => ChangeAccessibilityModifierAsync(context.Document, context.Span, accessibilityModifier, addVirtualModifier, cToken),
+				equivalenceKey: Resources.PX1077Fix);
+
+			context.RegisterCodeFix(codeFixAction, diagnostic);
+			return Task.CompletedTask;
+		}
+
+		private static readonly List<SyntaxKind> ModifiersToBeRemoved =
+		[
+			SyntaxKind.PrivateKeyword,
+			SyntaxKind.PublicKeyword,
+			SyntaxKind.ProtectedKeyword,
+			SyntaxKind.InternalKeyword,
+			SyntaxKind.VirtualKeyword
+		];
+
+		private static async Task<Document> ChangeAccessibilityModifierAsync(
+			Document document,
+			TextSpan span,
+			SyntaxKind accessibilityModifier,
+			bool addVirtual,
+			CancellationToken cancellationToken)
+		{
+			var root = await document
+				.GetSyntaxRootAsync(cancellationToken)
+				.ConfigureAwait(false);
+
+			if (root?.FindNode(span) is not MethodDeclarationSyntax method)
+			{
+				return document;
+			}
+
+			var newToken = SyntaxFactory.Token(accessibilityModifier);
+			var firstToken = method.Modifiers.FirstOrDefault();
+			var removeLeadingTriviaFromReturnType = firstToken == default;
```

Comment:
I think, the algorithm here will be clearer if you introduce a bool variable `hasModifiers`:
```cs
bool hasModifiers = firstToken != default;
bool removeLeadingTriviaFromReturnType = !hasModifiers ;
newToken = hasModifiers 
    ? newToken.WithTriviaFrom(firstToken)
    : newToken.WithLeadingTrivia(method.ReturnType.GetLeadingTrivia()); 
...
```

```

Reply:
Done.